### ISSUE TITLE
"Save"-button in code-editor can be misleading

### ISSUE THREAD

**Issue by JakobSchaeffeler** [ls1intum/Artemis#2543]
### Is your feature request related to a problem?
The online-editor has two options to *save* the code: "Save" and "Submit".
Some students didn't realize the difference between these two options and only clicked on "Save" once they finished their code.
Afterwards we got complaints of assessments not being corrected, but we couldn't find any code being submitted in the repositories.

### Describe the solution you'd like

It would be nice to add a warning, if the code is only saved but not submitted to make the difference clearer to students who have no experience with Artemis and Git, especially since tasks solved with the code-editor are normally not that complex and therefore "Submit" is the mostly wanted option. Of course any other solution which would make the difference more noticeable would be great as well.

### Describe alternatives you've considered

\---
### Additional context

\---

**Comment by krusche** [comment]
In my opinion, the word `Submit` is used consistently throughout the application (if you find a counter example, we are happy to change it) and the difference between `Save` and `Submit` should be clear. 

Sometimes, users do not want to submit, but only save their changes, because submitting them would have an effect, possibly a negative one on their grade. Another reason could be that they simply are not finished with the changes.

We could implement some kind of auto save mechanism, but this might not be what certain users want, as there are cases where you also would like to discard changes.

I could not imagine a warning that helps a couple of first time users (who do not think about the difference between `Save` and `Submit` and who might not even read a warning and simply click it away), but that would not disturb users who have understood the difference. However, I am open for suggestions, because usability for first time users is really important to me.

**Comment by FrankeLukas** [comment]
Imo (i.e. from a student view) it would make sense if the last saved state was automatically submitted once the due date is reached, but only if nothing was submitted before. (We don't want to override previous submissions.)

> I could not imagine a warning that helps a couple of first time users (who do not think about the difference between `Save` and `Submit` and who might not even read a warning and simply click it away), but that would not disturb users who have understood the difference. However, I am open for suggestions, because usability for first time users is really important to me.

Explaining this mechanic in the test course sounds better to me than displaying a warning.

**Comment by krusche** [comment]
I don’t think we should automatically submit something only in some cases. Then I would rather get rid of the save button at all. In other exercises we also do not have a save button. 

We could still automatically save the changes (as we do in other exercises), e.g. when students change the files, leave the page or after some time (e.g. 60s)

**Comment by FrankeLukas** [comment]
> Then I would rather get rid of the save button at all. In other exercises we also do not have a save button.

That sounds like the best solution then. I'd start working on this if there are no objections.

**Comment by anditurdiu** [comment]
Removing the save button can lead to performance issues though as every submit with trigger a build on the build agents.

Students are encoursged to frequently press save, as it  ensures that the user does not loose their changes on a reload or crash for example. Additionally, the save functionality is used for the exam mode if I am not mistaken (we cannot use submit for the autosave feature every 30 seconds due to performace reasons).

**Comment by jhainzlschmid** [comment]
I think we can close this issue. The save button has been removed and replaced by a autosave functionality
